src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +32,13 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0,numbers.size())
+               .mapToObj(i -> numbers.get(i) - (i % 2))
+               .filter(i -> i % 2 == 1)
+               .mapToInt(Integer::intValue)
+               .average()
+               .orElseThrow(() -> new NoSuchElementException("No element founded"));
```

Comment:
```suggestion
               .orElseThrow();
```

exception message "No element founded" gives no additional information to the exception that .orElseThrow() throws